include/appbase/application_base.hpp

```diff
@@ -379,7 +382,7 @@ class application_t : public application_base {
 
 private:
    inline static std::unique_ptr<application_t> app_instance;
-   executor_t executor_;
+   executor_t& executor_;
```

Comment:
Seems like it would be better to remove `executor_t& executor_` and just use `executor_ptr` instead.

Reply:
I think it is the same thing you are suggesting Kevin. I'd remove the `executor_` member and do:

```
   executor_t& executor() {
      return *static_cast<executor_t*>(executor_ptr.get());
   }
```

Reply:
Not sure I understand what is being suggested. Having the `executor_t& executor_` means I only need to do the cast once. If I don't have that, I need to add the cast in all of `post()`, `exec()`, `get_io_service()`, and `startup()` (and maybe in the ctor too).

Reply:
Is there a cost to the cast? I assumed the compiler would compile it away so that it is equivalent to what you have.

Reply:
I wasn't worried about the perf cost but rather copy/pasting the cast line around multiple places. But I can do that in a function as latest commit does